src/libopensc/card.c

```diff
@@ -952,6 +952,19 @@ int _sc_card_add_algorithm(sc_card_t *card, const sc_algorithm_info_t *info)
 	return SC_SUCCESS;
 }
 
+int _sc_card_add_generic(sc_card_t *card, unsigned int algorithm,
+			 unsigned int key_length, unsigned long flags)
```

Comment:
Oh, I remember that you should change this. The function is simply not *generic*. Either rename and export this to `_sc_card_add_symmetric_alg` as suggested by @dengert or export four dedicated functions for each algorithm (`_sc_card_add_des_alg`, _`sc_card_add_3des_alg`, `_sc_card_add_aes_alg`) possibly mapping to a single implementation of `_sc_card_add_symmetric_alg`.